Initial issue #12 commit

MikeT-gh opened the pull request:
Issue #12 

How-to guide describing how to install Landscape Server on Microsoft Azure, similar format to the Google Cloud how-to guide.

I have tried to make the guide so you can just copy and paste all of the commands in to get the Landscape Server up and running apart from the DNS steps.  I don't have much experience with Landscape Server apart from reading docs to get it installed.  Comments and feedback are very welcome.

rajannpatel reviewed (commented):
This is a wonderful contribution. I will get eyeballs on this so that we have parity with the [Google guide](https://ubuntu.com/landscape/docs/install-on-google-cloud), and also test it. Thanks for sharing this!

Hooloovoo commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,228 @@
+- If you’re using `cloud-init-quickstart.yaml`, run:
+```
+curl -s https://raw.githubusercontent.com/canonical/landscape-scripts/main/provisioning/cloud-init-quickstart.yaml -o cloud-init.yaml
+IMAGE_FAMILY=Canonical:0001-com-ubuntu-server-jammy:22_04-lts-gen2:latest
````
I think this should point to Canonical:0001-com-ubuntu-pro-jammy:pro-22_04-lts-gen2:latest to be equivalent to the https://ubuntu.com/landscape/docs/install-on-google-cloud it is based on.
See https://documentation.ubuntu.com/azure/en/latest/azure-how-to/instances/find-ubuntu-images/

YanisaHS replied:
Changed, see comment below

MikeT-gh commented:
Thank you @rajannpatel

MikeT-gh commented:
Updated PR.  Good point about the installation steps for the Azure CLI, have removed the steps and pointed it to the existing guide.  Thanks @Hooloovoo for the correction on the ubuntu images.  I have also added an extra note in the guide about an error users may come across when trying to create the VM, if the Azure subscription being used has not accepted the terms of the particular VM image being used.

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,228 @@
+```
+curl -s https://raw.githubusercontent.com/canonical/landscape-scripts/main/provisioning/cloud-init-quickstart-fips.yaml -o cloud-init.yaml
+IMAGE_FAMILY=Canonical:0001-com-ubuntu-server-focal:20_04-lts-gen2:latest
+```
````
Changed, see comment below

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,228 @@
+## Install and set up Microsoft Azure CLI
+
+### Install `Azure CLI`
+
```
Changed, see comment below

YanisaHS commented:
Hey @MikeT-gh! Just wanted to follow up. This PR is still on our radar - we're working on testing the steps with Azure, although it may take a while before we have more feedback for you. I'll keep you in the loop though! And thanks again for the helpful contribution!! :partying_face:

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+# How to install Landscape Server on Microsoft Azure
+
+This guide provides an example of how to install and set up your Landscape server on Microsoft Azure with [cloud-init](https://cloudinit.readthedocs.io/en/latest/). The instructions here can be used for both FIPS-hardened or non-hardened systems.
```
```suggestion
This guide provides an example of how to install and set up your Landscape server on Microsoft Azure with [cloud-init](https://cloudinit.readthedocs.io/en/latest/).
```
```suggestion
This guide provides an example of how to install and set up your Landscape server on Microsoft Azure with [cloud-init](https://cloudinit.readthedocs.io/en/latest/). The instructions here can be used for standard or FIPS-compliant deployments.
```

YanisaHS replied:
(You can ignore the first suggestion of deleting the last sentence.. not sure why it's there... :sweat_smile:)

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+This guide provides an example of how to install and set up your Landscape server on Microsoft Azure with [cloud-init](https://cloudinit.readthedocs.io/en/latest/). The instructions here can be used for both FIPS-hardened or non-hardened systems.
+
+> **For the most up-to-date documentation on Microsoft Azure, see [Microsoft Azure documentation](https://learn.microsoft.com/en-us/azure).
```
```suggestion
> **For the most up-to-date documentation on Microsoft Azure, see [Microsoft's Azure documentation](https://learn.microsoft.com/en-us/azure).
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+### Connect `Azure` with your Microsoft Azure account
+
+The Azure CLI's default authentication method for logins uses a web browser and access token to sign in.
```
```suggestion
The Azure CLI's default authentication method for logins uses a web browser and access token to sign in. To login, run:
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+### Connect `Azure` with your Microsoft Azure account
+
+The Azure CLI's default authentication method for logins uses a web browser and access token to sign in.
+
```
```suggestion
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+The Azure CLI's default authentication method for logins uses a web browser and access token to sign in.
+
+Run the `az login` command:
```
```suggestion
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+az login
+```
+
+If the Azure CLI can open your default browser, it initiates authorisation code flow and opens the default browser to load an Azure sign-in page.
````
```suggestion
If the Azure CLI can open your default browser, it will open the default browser and load an Azure sign-in page for you to sign in with your Azure account. Otherwise, it will instruct you to open a browser page and enter the code displayed in your terminal.
```

YanisaHS replied:
Just wanted to provide some additional information here.

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+If the Azure CLI can open your default browser, it initiates authorisation code flow and opens the default browser to load an Azure sign-in page.
+
+Sign in with your account credentials in the browser.
```
```suggestion
Sign in with your account credentials in the browser. For more information on signing in with the Azure CLI, see [Microsoft's documentation on signing in interactively with the Azure CLI](https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli-interactively).
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+### Create public-ip address resource
+
+Create static IP address in the `Landscape-rg` resource group:
```
```suggestion
Create a static IP address in the `Landscape-rg` resource group:
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+    --allocation-method Static
+```
+
+Output will be displayed in JSON format and will show static IP address (extract below):
````
```suggestion
Output will be displayed in JSON format and will show a static IP address:
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+
+Output will be displayed in JSON format and will show static IP address (extract below):
+```
+"ipAddress": "34.139.255.120",
````
```suggestion
"ipAddress": "34.XXX.XXX.XXX",
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+
+You’ll receive output similar to:
+```
+Server:		127.0.0.53
````
```suggestion
Server:		127.X.X.XX
```

MikeT-gh replied:
I have left the output from the nslookup command for the dns server and address.  As this is what most users output will be if using Ubuntu, as it points to the loopback address of 127.0.0.53 which systemd-resolved uses.

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+You’ll receive output similar to:
+```
+Server:		127.0.0.53
+Address:	127.0.0.53#53
````
```suggestion
Address:	127.X.X.XX#XX
```

MikeT-gh replied:
Refer to reply in above conversation.

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+   
+Non-authoritative answer:
+Name:	landscape.domain.com
+Address: 34.139.255.120
```
```suggestion
Address: 34.XXX.XXX.XXX
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+Before beginning the deployment process with cloud-init, you must choose which of the two cloud-init configuration automation templates you want to use. In the [Landscape Scripts](https://github.com/canonical/landscape-scripts) Github repository, there are two Landscape Quickstart cloud-init configuration templates: [`cloud-init-quickstart.yaml`](https://github.com/canonical/landscape-scripts/blo ...
+
+The `cloud-init-quickstart.yaml` template is designed for anyone, and the `cloud-init-quickstart-fips.yaml` is designed for FIPS compliant deployments of Landscape Server. For more information, see [how to install FIPS hardened Landscape Server](https://ubuntu.com/landscape/docs/install-fips-hardened-landscape-server).
```
```suggestion
The `cloud-init-quickstart.yaml` template is designed for anyone, and the `cloud-init-quickstart-fips.yaml` is designed for FIPS compliant deployments of Landscape Server. For more information, see [how to install FIPS-compliant Landscape Server](https://ubuntu.com/landscape/docs/install-fips-hardened-landscape-server).
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+    --image $IMAGE_FAMILY \
+    --size Standard_D2s_v3 \
+    --admin-username azureuser \
+    --assign-identity \
```
Can the `--assign-identity` parameter just be removed?

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+	--priority 100
+```
+
+It takes a few minutes to create the VM and supporting resources.
````
```suggestion
It usually takes a few minutes to create the VM and supporting resources.
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+It takes a few minutes to create the VM and supporting resources.
+
+> [!NOTE]
+When creating the VM an error may occur with the code `MarketplacePurchaseEligibilityFailed`. This error indicates that before the subscription can use this image, you need to accept the legal terms of the image. Viewing and accepting the terms can be done via the Azure CLI. Refer to Microsoft Azure documentation [az vm image terms](https://learn.microsoft.com/en-us/cli/azure/vm/image/terms).
```
```suggestion
When creating the VM, an error may occur with the code `MarketplacePurchaseEligibilityFailed`. This error indicates that before the subscription can use this image, you need to accept the legal terms of the image. You can view and accept the terms via the Azure CLI. Refer to [Microsoft's documentation on image terms](https://learn.microsoft.com/en-us/cli/azure/vm/image/terms).
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+ssh azureuser@{landscape.domain.com} 'tail -f /var/log/cloud-init-output.log'
+```
+
+A reboot may be required during the cloud-init process. If a reboot is required, you’ll receive the following output:
````
```suggestion
A reboot may be required during the cloud-init process. If a reboot is required, you’ll receive output similar to:
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
````diff
@@ -0,0 +1,205 @@
+ssh azureuser@{landscape.domain.com} 'tail -f /var/log/cloud-init-output.log'
+```
+
+Wait until the cloud-init process is complete. When it’s complete, you’ll receive the following line similar to this:
````
```suggestion
Wait until the cloud-init process is complete. When it’s complete, you’ll receive output similar to:
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,205 @@
+
+## Configure Landscape
+
+1. Navigate to the Landscape dashboard by entering the FQDN of the Landscape VM into a browser window
```
```suggestion
1. Navigate to the Landscape web portal by entering the FQDN of the Landscape VM into a browser window
```

YanisaHS replied:
We've changed to refer to it as the "web portal" now :smiley:

YanisaHS reviewed (changes requested):
Hi @MikeT-gh! Thanks so much for the AWESOME contribution!!!! It looks really good! I know you've waited a long time for this review, and I definitely appreciate your patience. :smiley: 

I've left some comments for you to address. Please feel free to tag me with any questions/comments, or otherwise, just let me know when you're ready for me to re-review.

Just to give you a heads up - I will be out of office until May 28. If you review any of these changes before then, I'll respond when I'm back in office. :smiley:

MikeT-gh commented:
Thank you @YanisaHS, I appreciate how thorough your review was.

Updated PR.  There was one review item I didn't end up changing, I will add a note under the conversation above for that change.

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,202 @@
+**Contents:**
+
+- Install and set up Microsoft Azure CLI
+- Provision Azure resources and deploy
```
```suggestion
- Provision Azure resources
```

YanisaHS replied:
Since the deploying step is in the next section

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,202 @@
+Sign in with your account credentials in the browser. For more information on signing in with the Azure CLI, see [Microsoft's documentation on signing in interactively with the Azure CLI](https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli-interactively).
+
+
+## Provision Azure resources and deploy
```
```suggestion
## Provision Azure resources
```

YanisaHS commented on landscape/how-to/install-landscape-server-on-microsoft-azure.md:
```diff
@@ -0,0 +1,202 @@
+
+
+### Create VM
+
```
```suggestion

Note: Microsoft recommends not storing sensitive data in the `--custom-data` field. For more information, see [Microsoft's documentation on custom data and cloud-init for virtual machines](https://learn.microsoft.com/en-us/azure/virtual-machines/custom-data).

```

YanisaHS replied:
If you have any additional thoughts on how this information could be noted to the user, feel free to provide other suggestions. :) I just want to make sure the user isn't accidentally exposing sensitive data without realizing.

YanisaHS reviewed (changes requested):
Thanks again for the great work! Just a few comments I found on a second pass. :smiley:

MikeT-gh commented:
Thanks @YanisaHS, sorry for not replying sooner.  I was looking into the possibility of the cloud-init script being able to pull in sensitive data from Azure KeyVault.  But the way I was able to get it to work was convoluted and not very good for a how-to guide.

Updated PR. 

One additional thought, would it be worth adding a line noting "_You can manually configure the SMTP settings after cloud-init as well. For more information, see [Landscape's documentation on configuring Postfix for email](https://ubuntu.com/landscape/docs/quickstart-deployment#heading--configure-postfix-for-email)._" below line 126 just before the "_Create VM_" section?  Just to let the user know this can also be configured at a later stage.

YanisaHS reviewed (approved):
Looks great!

YanisaHS commented:
@MikeT-gh I don't think that line is necessary, although I appreciate your suggestion :smiley: I try not to send users to a specific place inside another [slightly] unrelated document like that so it doesn't create confusion. I may add that line in later if I add more documentation on the SMTP email configurations.

YanisaHS commented:
@MikeT-gh Thanks so much for the spectacular contribution!!!!! :partying_face: :balloon: :tada: I know this PR was a long process, but I very much appreciate your engagement and the great work you produced. Here is the live link: https://ubuntu.com/landscape/docs/install-on-microsoft-azure

If you have any feedback on your experience working on this PR and the Open Documentation Academy, feel free to leave a comment here.

I look forward to collaborating with you again sometime! :partying_face:

MikeT-gh commented:
Oh great. Thank you so much @YanisaHS, really appreciate your guidance and communication.  I ended up learning a lot throughout this process.  

The Documentation Office Hours on youtube have been really helpful too.
